src/CommunityToolkit.Maui.Markup.SourceGenerators/Extensions/NamespaceSymbolExtensions.cs

```diff
@@ -0,0 +1,178 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis;
+
+namespace CommunityToolkit.Maui.Markup.SourceGenerators;
+
+static class NamespaceSymbolExtensions
+{
+	public static IEnumerable<INamedTypeSymbol> GetNamedTypeSymbols(this INamespaceSymbol namespaceSymbol)
+	{
+		var stack = new Stack<INamespaceSymbol>();
+		stack.Push(namespaceSymbol);
+
+		while (stack.Count > 0)
+		{
+			var namespaceSymbolFromStack = stack.Pop();
+
+			foreach (var member in namespaceSymbolFromStack.GetMembers())
+			{
+				if (member is INamespaceSymbol memberAsNamespace)
+				{
+					stack.Push(memberAsNamespace);
+				}
+				else if (member is INamedTypeSymbol memberAsNamedTypeSymbol)
+				{
+					yield return memberAsNamedTypeSymbol;
+				}
+			}
+		}
+	}
+
+	public static string GetFullTypeString(this INamedTypeSymbol type) =>
+		$"{type.Name}{(type.NullableAnnotation == NullableAnnotation.Annotated ? "?" : string.Empty)}{type.TypeArguments.GetGenericTypeArgumentsString()}";
+
+	public static string GetGenericTypeArgumentsString(this IEnumerable<ITypeSymbol> typeArguments)
+	{
+		if (!typeArguments.Any())
+		{
+			return string.Empty;
+		}
+
+		StringBuilder result = new StringBuilder("<");
+
+		bool isFirstArgument = true;
+		foreach (ITypeSymbol typeArg in typeArguments)
+		{
+			if (isFirstArgument)
+			{
+				isFirstArgument = false;
+			}
+			else
+			{
+				result.Append(", ");
+			}
+
+			switch (typeArg)
+			{
+				case ITypeParameterSymbol typeParameterSymbol:
+					result.Append(typeParameterSymbol.Name);
+					break;
+				case INamedTypeSymbol namedTypeSymbol:
+					result.Append(namedTypeSymbol.GetFullTypeString());
+					break;
+				default:
+					//Maybe should handle this somehow, maybe inject SourceProductionContext and call ReportDiagnostic on it?
```

Comment:
Do we care if the switch lands in `default`? If don't let's just do nothing (:

Reply:
Unsure! I blindly copy/pasted this from your code in `CommunityToolkit.Maui.SourceGenerators` 😄